rclcpp/include/rclcpp/memory_strategy.hpp

```diff
@@ -101,6 +155,11 @@ class MemoryStrategy
   std::vector<rclcpp::subscription::SubscriptionBase::SharedPtr> subs;
   std::vector<rclcpp::service::ServiceBase::SharedPtr> services;
   std::vector<rclcpp::client::ClientBase::SharedPtr> clients;
+
+  std::vector<void *> subscription_handles;
```

Comment:
@dirk-thomas clarified that if the vector's capacity is smaller than the requested size, `std::vector.resize()` will allocate a block of new entries, often to the next power of two. So resize won't necessarily allocate every time it's called, but it may incur a large penalty when the capacity needs to grow.

But for our purposes this is fine since these vectors will usually store small numbers (<16).